url/resources/setters_tests.json

```diff
@@ -1571,6 +1571,33 @@
                 "pathname": "me@example.net"
             }
         },
+        {
+            "comment": "Special URLs cannot have their paths erased",
+            "href": "file:///some/path",
+            "new_value": "",
+            "expected": {
+                "href": "file:///",
+                "pathname": "/"
+            }
+        },
+        {
+            "comment": "Non-special URLs can have their paths erased",
+            "href": "foo://somehost/some/path",
+            "new_value": "",
+            "expected": {
+                "href": "foo://somehost",
+                "pathname": ""
+            }
+        },
+        {
+            "comment": "Path-only URLs cannot have their paths erased",
+            "href": "foo:/some/path",
+            "new_value": "",
+            "expected": {
+                "href": "foo:/",
+                "pathname": "/"
+            }
+        },
```

Comment:
Can we also add a test where the new `pathname` value is non-empty but rootless, like `"test"`? A root should be automatically created in that case.

Reply:
Could we also add a test setting a path on a URL without a host?  My initial late-night implementation of this change would've failed that test.  Like this:

+        {
+            "comment": "Non-special URLs can have their paths erased",
+            "href": "foo:///some/path",
+            "new_value": "",
+            "expected": {
+                "href": "foo://",
+                "pathname": ""
+            }
+        },